### PR TITLE
Adding Vector Search support in Elasticsearch

### PR DESCRIPTION
**Pull request by atarora**
1. Vector search supported now in Elasticsearch with Minilm + clip model.
2. The UI exposes 4 new ways to interact with vectors viz. Pure vector text search , Text Vector + BM25 search , pure Image vector Search , Image Vector + BM25 search
3. Vector search can be started with ./quickstart.sh -vector
4. Added Embedding service to support query encodings
Big Thanks to @mkr for a big support on this!

### CODE REVIEW

File: elasticsearch/elasticsearch.yml
Comment (risdenk) [inline elasticsearch/elasticsearch.yml]: Why is this necessary? Is this trying to index a single file thats huge?

https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-network.html suggests to really not do that.

If sending a smaller file does `indexing_pressure.memory.limit` need to be there?

File: index-vectors.sh
Comment (risdenk) [inline index-vectors.sh:6]: `mkdir -p DATA_DIR` would make this folder as needed.

File: elasticsearch/data/.keep
Comment (risdenk) [inline elasticsearch/data/.keep]: Why does this folder need to be here? Can't `index-vectors.sh` create it? I left a comment on how to do that.

File: elasticsearch/Dockerfile
Comment (risdenk) [inline elasticsearch/Dockerfile:2]: does the snapshot zip file need to be checked into the repo? There is no maven snapshot release that can be pulled down maybe?

File: data-encoder/ecommerce/vectors/products.py
Comment (risdenk) [inline data-encoder/ecommerce/vectors/products.py]: Commented out lines can be removed?

File: data-encoder/ecommerce/vectors/products.py
Comment (risdenk) [inline data-encoder/ecommerce/vectors/products.py]: https://docs.python.org/3/library/zipfile.html#zipfile.ZipFile.open

File: elasticsearch/Dockerfile
Comment (mkr) [reply elasticsearch/Dockerfile:2]: Currently, yes, as we don't publish any snapshot. I think we should absolutely wait for the first release of the ES Embeddings Rewriter before merging this PR. I'll mark it WIP.

File: elasticsearch/elasticsearch.yml
Comment (mkr) [reply elasticsearch/elasticsearch.yml]: Thanks, yes, it was a bit sloppy to increase limits instead of splitting the files to achieve smaller bulk indexing request. I've changed the indexing bash script and split the files now and have removed the limits.

File: index-vectors.sh
Comment (mkr) [reply index-vectors.sh:6]: Done

File: elasticsearch/data/.keep
Comment (mkr) [reply elasticsearch/data/.keep]: Done

**Review by risdenk — APPROVED** [review]

File: data-encoder/README.md
Comment (mkr) [inline data-encoder/README.md:35]: We still need a standard dependency file (e.g. a `requirements.txt` that you `pip install`, or a `pyproject.toml` that you `poetry update`) describing the required dependencies to run the Python code.

Comment (atarora) [reply data-encoder/README.md:35]: I added that , but looks like I forgot to add the file to vcs before pushing.

**Pull request merged**
